cmd/osmosisd/cmd/init.go

```diff
@@ -167,6 +169,18 @@ func InitCmd(mbm module.BasicManager, defaultNodeHome string) *cobra.Command {
 
 			tmcfg.WriteConfigFile(filepath.Join(config.RootDir, "config", "config.toml"), config)
 
+			// create environment file in default node home
+			if defaultNodeHome == app.DefaultNodeHome {
```

Comment:
q: what exactly does this check mean? I do not really understand.

Can you please add a comment explaining why we need it

Reply:
I think we should remove this condition because caller already use `DefaultNodeHome` so no need to check more